### DISCUSSION TITLE
How to add @hasRole directive in generated graphql?

### DISCUSSION THREAD

**Discussion by oscar-ko** [ent/ent#3968] [Q&A]
I use entgql.QueryField(), in []schema.Annotation to generate GraphQL schema. How to add @hasRole directive?

ent.graphqls

```
type Query{
     tasks() @hasRole(role: USER)
}
```

**Comment by crossworth** [comment]
Hi @oscar-ko 

Check the links below:
- https://github.com/ent/contrib/blob/eff33e4dca0b84429e6127dc0ec30d866d28895c/entgql/internal/todo/ent/schema/group.go#L53-L55
- https://pkg.go.dev/entgo.io/contrib/entgql#Directives

**Reply by oscar-ko** [reply]
@crossworth  Thank you very much.. it solves my problem...  ^_^

**Comment by syssam** [comment]
@crossworth how to add on the create or update input?
